File: 프로그래머스/1/118666. 성격 유형 검사하기/성격 유형 검사하기.cpp
#include <bits/stdc++.h>

using namespace std;
char personalityTypes[4][2] = {{'R', 'T'},{'C', 'F'},{'J', 'M'},{'A', 'N'} };
string solution(vector<string> survey, vector<int> choices) {
    string answer = "";
    int len = survey.size();
    int personality[4] = {0, };
    for(int i = 0 ; i < len ; i++) {
        if(survey[i] == "RT") {
            personality[0] += choices[i] - 4;
        } else if(survey[i] == "TR") {
            personality[0] -= choices[i] - 4;
        } else if(survey[i] == "CF") {
            personality[1] += choices[i] - 4;
        } else if(survey[i] == "FC") {
            personality[1] -= choices[i] - 4;
        } else if(survey[i] == "JM") {
            personality[2] += choices[i] - 4;
        } else if(survey[i] == "MJ") {
            personality[2] -= choices[i] - 4;
        } else if(survey[i] == "AN") {
            personality[3] += choices[i] - 4;
        } else if(survey[i] == "NA") {
            personality[3] -= choices[i] - 4;
        } 
    }
    
    for(int i = 0 ; i < 4 ; i++) {
        // cout << personality[i] << ' ';
        if(personality[i] <= 0) {
            answer.push_back(personalityTypes[i][0]);
        } else {
            answer.push_back(personalityTypes[i][1]);
        }
    }
    return answer;
}

File: 프로그래머스/1/118666. 성격 유형 검사하기/성격 유형 검사하기.js
function solution(survey, choices) {
    let scores = new Map();
    scores.set('R', 0);
    scores.set('T', 0);
    scores.set('C', 0);
    scores.set('F', 0);
    scores.set('J', 0);
    scores.set('M', 0);
    scores.set('A', 0);
    scores.set('N', 0);
    
    var answer = '';
    choices.forEach((choice, index) => {
        scores.set(survey[index][0], scores.get(survey[index][0]) + 4 - choice);
        scores.set(survey[index][1], scores.get(survey[index][1]) + choice - 4);
    })
    
    
    if(scores.get('R')>= 0) {
        answer += 'R';
    } else {
        answer += 'T';
    }
    
    if(scores.get('C') >= 0) {
        answer += 'C';
    } else {
        answer += 'F';
    }
    
    if(scores.get('J')>= 0) {
        answer += 'J';
    } else {
        answer += 'M';
    }
    
    if(scores.get('A') >= 0) {
        answer += 'A';
    } else {
        answer += 'N';
    }
    return answer;
}